packages/hardhat/hardhat.config.ts

```diff
@@ -67,6 +67,7 @@ const config: HardhatUserConfig = {
   networks: {
     hardhat: {
       initialBaseFeePerGas: 0,
+      gasPrice: 0,
```

Comment:
Why this ?

Reply:
My hardhat transactions were getting mined with a 1 gwei gas price. even with base fee per gas set to 0, so expect statements that compared eth balances before and after failed.  

Maybe its eip1559 transactions vs legacy transactions (where even if base fee is 0, if you use a gas price of 1, it will use that as a priority fee).

Changing this made every transaction have a gas price of 0 and eth balances were not impacted by tx fees.

Reply:
However as I noted in slack, it seems that solidity coverage plugin overrides the 0 gas price to 1 gwei, so coverage was broken.  Fixed this by fetching the gas consumed by transactions.

Given I made this second fix, its likely I can remove this, but ideally would be better to just force all tests to use a 0 gas price, including coverage, but I couldn't figure out how to do this.

Reply:
Actually looking at the tests, sometimes we need the tx receipt gas price trick sometimes we dont to have coverage pass. I have no idea why.

Example that doesn't need it:

```
      const depositorEthBalanceBefore = await provider.getBalance(depositor.address)
      const slippage = BigNumber.from(3).mul(BigNumber.from(10).pow(16))
      const limitPriceEthPerPowerPerp = squeethPrice.mul(one.sub(slippage)).div(one);

      await controller.connect(depositor).updateOperator(vaultId, controllerHelper.address);
      await (positionManager as INonfungiblePositionManager).connect(depositor).approve(controllerHelper.address, tokenId); 
      await controllerHelper.connect(depositor).closeShortWithUserNft({
        vaultId, 
        tokenId,
        liquidity: positionBefore.liquidity,
        liquidityPercentage: liquidityPercentage,
        wPowerPerpAmountToBurn: mintWSqueethAmount, 
        collateralToWithdraw: vaultBefore.collateralAmount, 
        limitPriceEthPerPowerPerp, 
        amount0Min: BigNumber.from(0), 
        amount1Min:BigNumber.from(0)
      })

      const positionAfter = await (positionManager as INonfungiblePositionManager).positions(tokenId);
      const vaultAfter = await controller.vaults(vaultId);
      const depositorEthBalanceAfter = await provider.getBalance(depositor.address)

      expect(positionAfter.liquidity.eq(BigNumber.from(0))).to.be.true
      expect(vaultAfter.shortAmount.eq(BigNumber.from(0))).to.be.true
      expect(vaultAfter.collateralAmount.eq(BigNumber.from(0))).to.be.true

      if(wPowerPerpAmountToWithdraw.lt(mintWSqueethAmount)) {
        const ethToBuySqueeth = (mintWSqueethAmount.sub(wPowerPerpAmountToWithdraw)).mul(squeethPrice).div(one); 
        const remainingETHFromLp = wethAmountToWithdraw.sub(ethToBuySqueeth);

        expect(Number(depositorEthBalanceAfter.sub(depositorEthBalanceBefore).sub(vaultBefore.collateralAmount.add(remainingETHFromLp)).div(one).toString()) <= 0.01).to.be.true
      }
      else if (wPowerPerpAmountToWithdraw.gt(mintWSqueethAmount)) {
        const wPowerPerpAmountToSell = wPowerPerpAmountToWithdraw.sub(mintWSqueethAmount);
        const ethToGet = wPowerPerpAmountToSell.mul(squeethPrice).div(one);

        expect(Number(depositorEthBalanceAfter.sub(depositorEthBalanceBefore).sub(vaultBefore.collateralAmount.add(ethToGet).add(wethAmountToWithdraw)).div(one).toString()) <= 0.01).to.be.true
      }

```